src/MapPoint.cc

```diff
@@ -577,4 +582,76 @@ void MapPoint::UpdateMap(Map* pMap)
     mpMap = pMap;
 }
 
+void MapPoint::PreSave(set<KeyFrame*>& spKF,set<MapPoint*>& spMP)
+{  
+    
+    mBackupReplacedId = -1;
+    if(mpReplaced && spMP.find(mpReplaced) != spMP.end())
+        mBackupReplacedId = mpReplaced->mnId;
+
+    /*
+    mBackupObservationsId1.clear();
+    mBackupObservationsId2.clear();
+
+    // Save the id and position in each KF who view it
+    for(std::map<KeyFrame*,std::tuple<int,int> >::const_iterator it = mObservations.begin(), end = mObservations.end(); it != end; ++it)
+    {
+        KeyFrame* pKFi = it->first;
+        if(spKF.find(pKFi) != spKF.end())
+        {
+            mBackupObservationsId1[it->first->mnId] = get<0>(it->second);
+            mBackupObservationsId2[it->first->mnId] = get<1>(it->second);
+        }
+        else
+        {
+            EraseObservation(pKFi);
+        }
```

Comment:
I comment out line 605-608 when recover mObservations from backup, probably this the bug

Reply:
> I comment out line 605-608 when recover mObservations from backup, probably this the bug

Hello,
Could we add wechat (william_hjx)? I have few questions about the commented lines. Thx!